### PR TITLE
[libc++][C++03] Update include guards

### PR DESCRIPTION
**Pull request by philnik777**
This patch updates the includes guards of the C++ 03 headers.

This is part of https://discourse.llvm.org/t/rfc-freezing-c-03-headers-in-libc.

### CODE REVIEW

**Comment by philnik777** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/109001?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#109002** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/109002?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#109001** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/109001?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#109000** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/109000?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#108999** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/108999?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @philnik777 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: libcxx/include/__cxx03/__algorithm/pstl.h [hunk 1]
```diff
@@ -41,7 +41,7 @@ template <class _ExecutionPolicy,
           enable_if_t<is_execution_policy_v<_RawPolicy>, int> = 0>
 [[nodiscard]] _LIBCPP_HIDE_FROM_ABI bool
 any_of(_ExecutionPolicy&& __policy, _ForwardIterator __first, _ForwardIterator __last, _Predicate __pred) {
-  _LIBCPP_REQUIRE_CPP17_FORWARD_ITERATOR(_ForwardIterator, "any_of requires a ForwardIterator");
+  _LIBCPP___CXX03_REQUIRE_CPP17_FORWARD_ITERATOR(_ForwardIterator, "any_of requires a ForwardIterator");
```
Comment (ldionne) [inline libcxx/include/__cxx03/__algorithm/pstl.h:44]: Oops!

**Review by ldionne — COMMENTED** [review]
This looks good to me in principle, however the patch itself has a few incorrect renamings. One thing you could do is use `git diff --stat` to confirm that all the changed files have exactly 3 lines changed in them. A few headers like the C compat headers might need additional renames, that needs to be investigated.

**Review by ldionne — APPROVED** [review]

**Review by ldionne — CHANGES_REQUESTED** [review]
Oops, there are still many instances of `_LIBCPP___CXX03_REQUIRE_`

**Review by ldionne — APPROVED** [review]

**Comment by philnik777** [comment]
### Merge activity

* **Dec 9, 11:33 AM EST**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/llvm/llvm-project/109001).

**Pull request merged**